Platform/demo/hal/demo_hal.c

```diff
@@ -54,7 +54,7 @@ void HAL_sys_getTime(uint32_t *unix_timestamp) {
 SAT_returnState HAL_hk_report(uint8_t sid, void *output) {
   switch (sid) {
     case BATTERY_1:
-      if ((sizeof((char *) output) + 1) > csp_buffer_data_size()) {
+      if ((sizeof((char *) output) + 1) > csp_buffer_data_size() - 2) {
```

Comment:
This still doesn't quite check for the potential problem. What we want is to make sure that the data field of the CSP packets (minus 2) is large enough to hold the battery struct. Something like this:
```
if ( sizeof(HK_battery) > (csp_buffer_data_size() - 2)) { // error }
```

Reply:
In fact, I'm pretty sure that ` ((sizeof((char *) output) + 1) > csp_buffer_data_size() - 2) ` is a tautology since ((sizeof((char *) output)) is by definition equal to csp_buffer_data_size()